pkg/controller/sparkapplication/controller.go

```diff
@@ -139,6 +140,7 @@ func newSparkApplicationController(
 		enableUIService:          enableUIService,
 		disableExecutorReporting: disableExecutorReporting,
 		executorsProcessingLimit: executorsProcessingLimit,
+		submissionCache:          NewSubmissionCache(5 * time.Minute),
```

Comment:
TTL as 5 minutes, arbitrary number

Reply:
should we make it configurable?

Reply:
I would say no,
I don't think this value need to be change externally